src/ui/hooks/useMetadata.ts

```diff
@@ -75,18 +75,6 @@ function validate(metadata: Abi | undefined, { isWasmRequired }: Options): Valid
     };
   }
 
-  const wasm = metadata.info.source.wasm;
-  const isWasmEmpty = wasm.isEmpty;
-  const isWasmInvalid = !WebAssembly.validate(wasm);
-
-  if (isWasmRequired && (isWasmEmpty || isWasmInvalid)) {
```

Comment:
If we just go with removal, we should reflect that as well in the `Options`.

https://github.com/paritytech/contracts-ui/blob/66a276e314ee5c2da85277b7a3eaed7a9667f9aa/src/ui/hooks/useMetadata.ts#L11-L13

Reply:
yup, didn't finish this yet. marked as draft now.

Reply:
it's a case of bad naming here. `isWasmRequired` means you are calling `instantiate_with_code`

this PR only affects the file input validation. nothing else needs to be removed actually